internal/csi-addons/rbd/volumegroup.go

```diff
@@ -117,6 +117,19 @@ func (vs *VolumeGroupServer) CreateVolumeGroup(
 
 	log.DebugLog(ctx, "VolumeGroup %q has been created: %+v", req.GetName(), vg)
 
+	// extract the flatten mode
+	flattenMode, err := getFlattenMode(ctx, req.GetParameters())
+	if err != nil {
+		return nil, err
+	}
+	// Flatten the image if the flatten mode is set to FlattenModeForce
+	// before adding it to the volume group.
+	for _, v := range volumes {
+		err = v.HandleParentImageExistence(ctx, flattenMode)
+		if err != nil {
+			return nil, getGRPCError(err)
```

Comment:
include a bit more details in the error maybe? Mention the VolumeGroup and Volume?

Reply:
added volumegroup, volume details are already returned by HandleParentImageExistence